### PR TITLE
feat(typescript-estree): add parserOption to turn on debug logs

### PR DESCRIPTION
**Pull request by bradzacher**
From the cli you can do `DEBUG=typescript-eslint:*,eslint:*` and it'll turn on all debug logging.
However, when debugging things running in an eslint IDE extension, it's hard to get debug information, which can make it really hard to trace bugs.
It's relatively easy for maintainers to go and edit the files in `node_modules`, but it's not really a good experience to ask issue reporters to do the same to give us logs.

This just adds the following parserOption
`debugLevel?: boolean | ('typescript-eslint' | 'eslint' | 'typescript')[]`
Which lets you turn on debug logging for all of the relevant modules. I opted for an array instead of a number so it's more explicit and easier to fine tune, because each of the modules can dump a tonne of logs.

### CODE REVIEW

File: packages/typescript-estree/src/parser-options.ts [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+type DebugModule = 'typescript-eslint' | 'eslint' | 'typescript';
+/**
+ * For convenience:
+ * - true === ['typescript-eslint']
+ * - false === []
+ *
+ * An array of modules to turn explicit debugging on for.
+ * - 'typescript-eslint' is the same as setting the env var `DEBUG=typescript-eslint:*`
+ * - 'eslint' is the same as setting the env var `DEBUG=eslint:*`
+ * - 'typescript' is the same as setting `extendedDiagnostics: true` in your tsconfig compilerOptions
```
Comment (armano2) [inline packages/typescript-estree/src/parser-options.ts]: can you add this note somewhere in docs?

Comment (bradzacher) [reply packages/typescript-estree/src/parser-options.ts]: I'm working on updating the ts-estree docs, as they're rather outdated.
I'll submit the readme update as a separate PR.

**Comment by bradzacher** [comment]
The problem is that it's pretty hard to test this, as the code is buried pretty deep in the stack.
I guess I could mock the `debug` module and assert `debug.enable` is called?

File: packages/typescript-estree/README.md [hunk 2]
```diff
@@ -29,54 +29,140 @@ yarn add -D @typescript-eslint/typescript-estree
...
+   */
+  debugLevel?: boolean | ('typescript-eslint' | 'eslint' | 'typescript')[];
 
-  // create a top-level comments array containing all comments
-  comment: false,
+  /**
+   * Cause the parser to error if it encounters an unknown AST node type (useful for testing).
+   * This case only usually occurs when TypeScript releases new features.
+   */
+  errorOnUnknownASTType?: boolean;
```
Comment (armano2) [inline packages/typescript-estree/README.md]: ye finally 👍

**Pull request merged**